src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7382,4 +7382,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_DeconstructVariableCannotBeByRef" xml:space="preserve">
     <value>A deconstruction variable cannot be declared as a ref local</value>
   </data>
+  <data name="ERR_NonConstantConversionInConstantPattern" xml:space="preserve">
+    <value>Cannot implicitly convert type '{0}' to type '{1}' in constant pattern using non-constant conversion</value>
```

Comment:
>non-constant conversion [](http://example.com/codeflow?start=88&length=23)

I don't think there is a concept of "non-constant conversion" or "constant conversion" in the language. Since the error is reported specifically for user-defined conversions, we should consider saying exactly that: "Cannot use user-defined conversion for pattern expression." or some thing similar.

Reply:
I have clarified the message as per your recommendation.